store/db.go

```diff
@@ -44,39 +48,49 @@ func NewDB(dbdir string) (*DB, error) {
 
 func (db *DB) Open() error {
 	// take a lock on db dir
-	if db.lock != nil {
-		panic("cas db lock already gained")
-	}
 	dl, err := lock.ExclusiveLock(db.dbdir, lock.Dir)
 	if err != nil {
 		return err
 	}
-	db.lock = dl
 
 	sqldb, err := sql.Open("ql", filepath.Join(db.dbdir, DbFilename))
 	if err != nil {
 		dl.Close()
 		return err
 	}
+
+	// (XXX): flock is already granted, the mutex
+	// here is merely for making race detector happy.
+	db.Lock()
```

Comment:
https://en.wikipedia.org/wiki/Comment_%28computer_programming%29
XXX - warn other programmers of problematic or misguiding code